MapleServer2/Data/Static/MasteryMetadataStorage.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Maple2Storage.Types.Metadata;
+using MapleServer2.Constants;
+using ProtoBuf;
+
+namespace MapleServer2.Data.Static
+{
+    public static class MasteryMetadataStorage
+    {
+        private static readonly Dictionary<int, MasteryMetadata> masteries = new Dictionary<int, MasteryMetadata>();
+
+        static MasteryMetadataStorage()
+        {
+            using FileStream stream = File.OpenRead($"{Paths.RESOURCES}/ms2-mastery-metadata");
+            List<MasteryMetadata> masteryList = Serializer.Deserialize<List<MasteryMetadata>>(stream);
+            foreach (MasteryMetadata mastery in masteryList)
+            {
+                masteries[mastery.Type] = mastery;
+            }
+        }
+
+        public static List<int> GetMasteryTypes()
+        {
+            return new List<int>(masteries.Keys);
+        }
+
+        public static MasteryMetadata GetMastery(int type)
+        {
+            return masteries.GetValueOrDefault(type);
+        }
+        
+        public static int? GetMasteryTierRewardItemId(int type, int grade)
+        {
+            MasteryMetadata mastery = GetMastery(type);
+            if (mastery != null)
+            {
+                return mastery.Grades[grade - 1].RewardJobItemID;
+            }
+
+            return null;
+        }
```

Comment:
Prefer handling this in whatever method calls this, unless it it called at least 2 times.